### PR TITLE
Feat/rollback

### PR DESCRIPTION
**Pull request by sh-cha**
- rollup sync from initia-l1, celestia

### CODE REVIEW

File: rollupsync/provider/celestia.go
Comment (beer-1) [inline rollupsync/provider/celestia.go]: how about `BatchChainHeight`?

File: config/toml.go
Comment (beer-1) [inline config/toml.go]:
```toml
// `batch_rpc` can be empty, if the `batch_chain` is "l1"
```

File: config/toml.go
Comment (beer-1) [inline config/toml.go]: `inita1~` or `celestia1~` address depends on batch_chain config.

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: `GetRollupSyncBatchChainHeight()`?

Comment (beer-1) [reply rollupsync/sync.go]: is it okay to ignore error?

File: rollupsync/provider/utils.go
Comment (beer-1) [inline rollupsync/provider/utils.go]: Seems good to move to config

File: rollupsync/const.go
Comment (beer-1) [inline rollupsync/const.go]: This seems too small, so good to use 10 or bigger number or make this also configurable

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: maybe we need some way to stop this rollup sync process when this go thread met error by ctx <- done?  or quite

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: This too

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: here too

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: panic to stop

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: panic to stop

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: maybe good to stop with error not panic.

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: This too

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: BatchChainHeight

File: rollupsync/types/channel.go
Comment (beer-1) [inline rollupsync/types/channel.go]: BatchChainHeight

File: rollupsync/types/channel.go
Comment (beer-1) [inline rollupsync/types/channel.go]: BatchChainHeight

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go:176]: chunks should be reset when decompression succeed

File: rpc/core/types/responses.go
Comment (beer-1) [inline rpc/core/types/responses.go]: raw commit bytes

File: state/store.go
Comment (beer-1) [inline state/store.go]: // overwrite validator set if the executor is changed

File: rollupsync/sync.go
Comment (beer-1) [inline rollupsync/sync.go]: don't need to save seen commit here, seen commit is used only in consensus.

File: state/execution.go
Comment (beer-1) [inline state/execution.go]: old validators can be multiple, so maybe we need to handle that case here.

**Review by beer-1 — COMMENTED** [review]
LGTM 👍, seems need minor changes to make const to configurable.

need change in `checkExecutorChanged`

File: rollupsync/sync.go
Comment (sh-cha) [reply rollupsync/sync.go]: it is to start from scratch if there is no saved height

File: config/config.go
Comment (beer-1) [inline config/config.go]: seems condition is wrong

File: config/config.go
Comment (beer-1) [inline config/config.go:1005]: This function should be called in L132

File: rollupsync/utils.go
Comment (beer-1) [inline rollupsync/utils.go]: debug print

File: rollupsync/utils.go
Comment (beer-1) [inline rollupsync/utils.go]: debug print

**Review by beer-1 — APPROVED** [review]
https://github.com/initia-labs/cometbft/pull/2/files

have made some cleanup pr, so good to merge after merge the one

**Pull request merged**
